iree/hal/vulkan/api.h

```diff
@@ -89,12 +96,66 @@ iree_hal_vulkan_syms_release(iree_hal_vulkan_syms_t* syms);
 
 #endif  // IREE_API_NO_PROTOTYPES
 
+//===----------------------------------------------------------------------===//
+// iree::hal::vulkan Extensibility Util
+//===----------------------------------------------------------------------===//
+
+#ifndef IREE_API_NO_PROTOTYPES
+
+// Gets the names of the Vulkan instance extensions needed to create a driver
+// with |iree_hal_vulkan_driver_create_using_instance| and |options|.
+//
+// |out_count| is a pointer to an integer related to the number of extensions.
+// |out_extensions| is either nullptr or a pointer to an array to be filled with
+// the required Vulkan driver extensions.
+IREE_API_EXPORT iree_status_t IREE_API_CALL
+iree_hal_vulkan_get_required_instance_extensions(
```

Comment:
since all of the signatures are the same we should enum it as a iree_hal_vulkan_extensibility_set_t enum with INSTANCE_REQUIRED/INSTANCE_OPTIONAL/etc instead - then there's only one place to do the validation and less code exposed.